NebulaPatcher/Patches/Dynamic/EnemyUnitComponent_Patch.cs

```diff
@@ -0,0 +1,400 @@
+    }
+
+    [HarmonyPostfix]
+    [HarmonyPatch(nameof(EnemyUnitComponent.SensorLogic_Ground))]
+    static void SensorLogic_Ground_Postfix(ref EnemyUnitComponent __instance, ref EnemyData enemy, PlanetFactory factory, int hatred_max_sense, EAggressiveLevel aggressiveLevel)
+    {
+        if (!Multiplayer.IsActive)
+        {
+            return;
+        }
+
+        var planetId = __instance.planetId;
+        if (!Multiplayer.Session.Combat.ActivedPlanets.Contains(planetId))
+        {
+            return;
+        }
+
+        var players = Multiplayer.Session.Combat.Players;
+        for (var i = 0; i < players.Length; i++)
+        {
+            if (players[i].planetId != planetId)
+            {
+                continue;
+            }
+            // Skip assaults.targetObject for now
+
+            var num = 1.0f;
+            switch (aggressiveLevel)
+            {
+                case EAggressiveLevel.Dummy:
+                    return;
+                case EAggressiveLevel.Passive:
+                    return;
+                case EAggressiveLevel.Torpid:
+                    num = 0.85f;
+                    break;
+                case EAggressiveLevel.Normal:
+                    num = 1.0f;
+                    break;
+                case EAggressiveLevel.Sharp:
+                    num = 1.2f;
+                    break;
+                case EAggressiveLevel.Rampage:
+                    num = 1.4f;
+                    break;
+            }
+
+            var prefabDescByModelIndex = SpaceSector.PrefabDescByModelIndex;
+            var prefabDesc = prefabDescByModelIndex[enemy.modelIndex];
+            var num2 = prefabDesc.unitSensorRange * num;
+            var num4 = num2 * 1.45f + factory.skillSystem.playerEnergyShieldRadius;
+            var num6 = (float)enemy.pos.x;
+            var num7 = (float)enemy.pos.y;
+            var num8 = (float)enemy.pos.z;
+
+
+            // Find the nearest alive player to enemy unit
+            var index = 0;
+            var distance = float.MaxValue;
+            var playerId = -1;
+            for (var j = 0; j < Multiplayer.Session.Combat.Players.Length; j++)
+            {
+                ref var player = ref Multiplayer.Session.Combat.Players[j];
+                if (player.planetId != planetId || !player.isAlive)
+                {
+                    continue;
+                }
+                var dx = player.position.x - num6;
+                var dy = player.position.y - num7;
+                var dz = player.position.z - num8;
+                var sqtDist = dx * dx + dy * dy + dz * dz;
+                if (sqtDist < distance)
+                {
+                    index = j;
+                    distance = sqtDist;
+                    playerId = player.id;
+                }
+            }
+            if (playerId == -1)
+            {
+                // If there is no alive player, return the original value
+                return;
+            }
```

Comment:
Minor nitpick, could be simplified to:
```suggestion
            // If there is no alive player, return the original value
            if (playerId == -1) return;
```